src/Reflection/ReflectionFunctionAbstract.php

```diff
@@ -97,13 +101,17 @@ public function getNumberOfParameters()
      */
     public function getNumberOfRequiredParameters()
     {
-        return count(array_filter($this->parameters, function (ReflectionParameter $p) {
-            return !$p->isOptional();
-        }));
+        return count(array_filter(
```

Comment:
Is this API in line with the other optional parameters API you wrote before?
As far as I understand, in your case, given following method:

```php
public function foo($bar, $baz = null, $tab, $taz = null) {}
```
the number of optional parameters is 1